tools/autograd/gen_autograd_functions.py

```diff
@@ -21,6 +21,7 @@
   ${will_release_variables}
   ${saved_variables}
   ${saved_list_sizes}
+  bool released = false;
```

Comment:
naming this `released` doesn't seem like the best idea because this will cause a problem with any function that tries to capture an argument named `released`.

Reply:
Also, can we conditionally generate this variable, like we do with every other variable in autograd Functions?

Reply:
I also think there are 2 more subtle issues with this:
1) why is this global on the function?  This would prevent us from conditionally unpacking the variables we need if we only need to compute some subset of the backwards, for no good reason.  We could just track this per TensorList, like we do with the size.
2) Speaking of the size, doesn't that solve your disambiguate problem?  Like, you know the size at the start and can check the later size, to figure out if something was reset.  I guess there is a judgement call on whether you should error if the TensorList that was captured was empty, but that seems like we can just do the same thing we do with Tensors (what happens if we capture an undefined tensor?).

Reply:
1. You can unpack some of the variable, flag would be set only after release_variables() will be called.
2. We do not hold TensorList, instead we hold vector

Reply:
Name has been changed to variables_were_released_

Reply:
are your responses "1." and "2." responses to my questions "1." and "2."?  I don't understand how they address the questions if so.  For example, with 2., I understand that you don't literally hold the "TensorList"; the question is about distinguishing between whether the thing you saved was empty to start or it was wasn't, but was released.